src/test/Dashboard/Charts/ChartCard.test.tsx

```diff
@@ -0,0 +1,229 @@
+  defaultServices.settings
+);
+const mockChartContext = {
+  controller: mockController,
+};
+
+describe('<ChartCard />', () => {
+  it('renders correctly', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.READY));
+
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <NotificationsContext.Provider value={NotificationsInstance}>
+            <ChartContext.Provider value={mockChartContext}>
+              <Provider store={store}>
+                <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('with-content');
+  });
+
+  it('renders loading state correctly', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.UNKNOWN));
+
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <NotificationsContext.Provider value={NotificationsInstance}>
+            <ChartContext.Provider value={mockChartContext}>
+              <Provider store={store}>
+                <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('loading-view');
+  });
+
+  it('renders empty state correctly', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    let tree;
+    await act(async () => {
+      tree = renderer.create(
+        <ServiceContext.Provider value={defaultServices}>
+          <NotificationsContext.Provider value={NotificationsInstance}>
+            <ChartContext.Provider value={mockChartContext}>
+              <Provider store={store}>
+                <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+              </Provider>
+            </ChartContext.Provider>
+          </NotificationsContext.Provider>
+        </ServiceContext.Provider>
+      );
+    });
+    expect(tree.toJSON()).toMatchSnapshot('empty-state');
+  });
+
+  it('renders empty state with information and action button', async () => {
+    jest.spyOn(mockController, 'attach').mockReturnValue(of(ControllerState.NO_DATA));
+
+    renderChartCard(
+      <ChartCard theme={'light'} chartKind={'CPU Load'} duration={120} period={10} span={6} dashboardId={0} />
+    );
+
+    expect(screen.getByText('CPU Load')).toBeInTheDocument();
+    expect(screen.getByText('CHART_CARD.NO_RECORDING.TITLE')).toBeInTheDocument();
```

Comment:
Hmmm a little odd here. I thought the actual value must be showing instead of keys?

Reply:
Same other aria-labels too...

Reply:
```log
    console.warn
      react-i18next:: You will need to pass in an i18next instance by using initReactI18next

      106 |
      107 | export const ChartCard: React.FC<ChartCardProps> = (props) => {
    > 108 |   const [t] = useTranslation();
          |                             ^
      109 |   const serviceContext = React.useContext(ServiceContext);
      110 |   const controllerContext = React.useContext(ChartContext);
      111 |   const history = useHistory();

      at warn (node_modules/react-i18next/dist/commonjs/utils.js:22:31)
      at warnOnce (node_modules/react-i18next/dist/commonjs/utils.js:35:8)
      at useTranslation (node_modules/react-i18next/dist/commonjs/useTranslation.js:44:25)
      at ChartCard (src/app/Dashboard/Charts/ChartCard.tsx:108:29)
      at renderWithHooks (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:5552:18)
      at mountIndeterminateComponent (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:7918:13)
      at beginWork (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:9019:16)
      at performUnitOfWork (node_modules/react-test-renderer/cjs/react-test-renderer.development.js:12649:12)

    console.warn
      react-i18next:: You will need to pass in an i18next instance by using i18nextReactModule
```

Reply:
Ahh I figured out that `Common.tsx` initialized the test i18n module. So, without importing it, this happens. Maybe we can move the `renderChartCard` to `Common.tsx`?

Reply:
Makes sense, though I don't think `renderChartCard` should go directly into `Common`. Maybe there is some more generic refactoring that I can apply first so that the function can be reused and is actually "common". It is pretty much like all those others, except with the chart controller added.